src/main/kotlin/com/buildit/bookit/v1/booking/dto/Booking.kt

```diff
@@ -22,31 +29,23 @@ data class BookingRequest(
     val end: LocalDateTime?
 )
 
-const val MASKED_STRING = "**********"
-fun maskSubjectIfOtherUser(booking: Booking, otherUser: UserPrincipal): Booking =
-    when {
-        booking.user.externalId != otherUser.subject -> booking.copy(subject = MASKED_STRING)
-        else -> booking
-    }
-
 /**
  * Booking response
  */
+@Entity
 data class Booking(
-    val id: String,
-    val bookableId: String,
+    @ManyToOne(optional = false)
+    val bookable: Bookable,
+    @Column(nullable = false)
     val subject: String,
-    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
+    @Column(nullable = false) @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
     val start: LocalDateTime,
-    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
+    @Column(nullable = false) @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
     val end: LocalDateTime,
-    val user: User
-)
-
-data class User(
-    val id: String,
-    val name: String,
-    val externalId: String
+    @ManyToOne(optional = false)
+    val user: User,
+    @Id @GeneratedValue(generator = "uuid2") @GenericGenerator(name = "uuid2", strategy = "uuid2")
+    val id: String? = null
```

Comment:
generally moved optional IDs to end

Reply:
fair point... i'll have a looksie as to what it actually does

Reply:
```
Hibernate: drop table bookable if exists
Hibernate: drop table booking if exists
Hibernate: drop table location if exists
Hibernate: drop table user if exists
Hibernate: create table bookable (id varchar(255) not null, closed boolean not null, reason varchar(255) not null, name varchar(255) not null, location_id varchar(255) not null, primary key (id))
Hibernate: create table booking (id varchar(255) not null, end timestamp not null, start timestamp not null, subject varchar(255) not null, bookable_id varchar(255) not null, user_id varchar(255) not null, primary key (id))
Hibernate: create table location (id varchar(255) not null, name varchar(255) not null, time_zone varchar(255) not null, primary key (id))
Hibernate: create table user (id varchar(255) not null, external_id varchar(255) not null, family_name varchar(255) not null, given_name varchar(255) not null, primary key (id))
Hibernate: alter table bookable add constraint UK_6ietqnvn35wqiovo8qrmwk2ne unique (location_id, name)
Hibernate: alter table location add constraint UK_sahixf1v7f7xns19cbg12d946 unique (name)
Hibernate: alter table user add constraint UK_4eu2tvn9rj53a93fufx7ayr20 unique (external_id)
Hibernate: alter table bookable add constraint FK1dc19oypan7r3ry418vn2h5sc foreign key (location_id) references location
Hibernate: alter table booking add constraint FKtbhx4q4q4m1kv4muhynct3p81 foreign key (bookable_id) references bookable
Hibernate: alter table booking add constraint FKkgseyy7t56x7lkjgu3wah5s3t foreign key (user_id) references user
```

Reply:
so looks like it does the right thing i'd like to see from a schema perspective wrt order